providers/provider_default.go

```diff
@@ -90,6 +90,9 @@ func (p *ProviderData) GetLoginURL(redirectURI, state string) string {
 	params.Set("client_id", p.ClientID)
 	params.Set("response_type", "code")
 	params.Add("state", state)
+	if p.ProtectedResource != nil && p.ProtectedResource.String() != "" {
```

Comment:
This is already a `string` type and defaults to `""`.  You can simplify this logic and mirror this line's format: `if p.AcrValues != "" {`

Can you add a unit test for this addition?

Reply:
As of current `master`, this is of type `*url.URL`, so I need to verify that it is not `nil` before checking for emptyness. I'll try to add a unit test.

Reply:
Ack - you are right! I was looking at the `options` ProtectedResource that made it - thanks!

Reply:
OK - I had a chance to research this more:

I think since this is a flag that is documented as `Azure` only, it shouldn't be added on the default `GetLoginURL` method, it should be under `azure.go` (which means making a GetLoginURL under AzureProvider)

...but I see a commit from 5 years ago where it was added to the default `Redeem` even though this is only set by the AzureProvider and the AzureProvider implements its own `Redeem` method. 🤷‍♂️ 

Can you take a try at getting this completely Azure specific?

Otherwise we can do it as you've listed here (having it part of defaults for both `Redeem` & `GetLoginURL`) -- I'd just like to update the documentation for the `--resource` flag that it applies to all providers and remove the note that it is an Azure only flag.

Reply:
Ah I was not aware that the `--resource` flag was meant to be Azure specific. I moved the code into azure.go